Library/Homebrew/test/dev-cmd/determine-test-runners_spec.rb

```diff
@@ -0,0 +1,198 @@
+# typed: false
+# frozen_string_literal: true
+
+require "dev-cmd/determine-test-runners"
+require "cmd/shared_examples/args_parse"
+
+describe "brew determine-test-runners" do
+  after do
+    FileUtils.rm_f github_output
+  end
+
+  # TODO: Generate this dynamically based on our supported macOS versions.
+  let(:linux_runner) { "ubuntu-22.04" }
+  let(:all_runners) { ["11", "11-arm64", "12", "12-arm64", "13", "13-arm64", linux_runner] }
+  let(:intel_runners) { all_runners.reject { |r| r.end_with? "-arm64" } }
+  let(:arm64_runners) { all_runners - intel_runners }
+  let(:macos_runners) { all_runners - [linux_runner] }
+  # We need to make sure we write to a different path for each example.
+  let(:github_output) { "#{TEST_TMPDIR}/github_output#{DetermineRunnerTestHelper.new.number}" }
+  let(:ephemeral_suffix) { "-12345-1" }
+  let(:runner_env) do
+    {
+      "HOMEBREW_LINUX_RUNNER"  => linux_runner,
+      "HOMEBREW_LINUX_CLEANUP" => "false",
+      "GITHUB_RUN_ID"          => ephemeral_suffix.split("-").second,
+      "GITHUB_RUN_ATTEMPT"     => ephemeral_suffix.split("-").third,
+    }.freeze
+  end
+
+  it_behaves_like "parseable arguments"
+
+  it "fails without any arguments", :integration_test do
```

Comment:
Don't want to have more than one integration test per command: they are very slow. Instead, try to make some of the code below unit tests instead.

Reply:
Will refactor. Keeping them around during refactoring to make sure I don't break anything.

Reply:
All good 👍🏻